assets/js/modules/search-console/components/module/ModulePopularKeywordsWidget/index.js

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { numFmt } from '../../../../../util';
+import { MODULES_SEARCH_CONSOLE, STORE_NAME, DATE_RANGE_OFFSET } from '../../../datastore/constants';
+import PreviewTable from '../../../../../components/PreviewTable';
+import Link from '../../../../../components/Link';
+import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import TableOverflowContainer from '../../../../../components/TableOverflowContainer';
+import ReportTable from '../../../../../components/ReportTable';
+import SourceLink from '../../../../../components/SourceLink';
+import { isZeroReport } from '../../../util/is-zero-report';
+import { generateDateRangeArgs } from '../../../util/report-date-range-args';
+import Header from './Header';
+
+const { useSelect } = Data;
+
+function ModulePopularKeywordsWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
+	const {
+		data,
+		isLoading,
+		error,
+		serviceURL,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
+		const { startDate, endDate } = dateRangeDates;
+
+		const reportArgs = {
+			startDate,
+			endDate,
+			url: url || undefined,
+			dimensions: 'query',
+			limit: 10,
+		};
+
+		return {
+			data: store.getReport( reportArgs ),
+			isLoading: ! store.hasFinishedResolution( 'getReport', [ reportArgs ] ),
+			error: store.getErrorForSelector( 'getReport', [ reportArgs ] ),
+			serviceURL: store.getServiceReportURL( {
+				...generateDateRangeArgs( dateRangeDates ),
+				page: url ? `!${ url }` : undefined,
+			} ),
+		};
+	} );
+
+	if ( error ) {
+		return <WidgetReportError moduleSlug="search-console" error={ error } />;
+	}
+
+	if ( isZeroReport( data ) ) {
+		return <WidgetReportZero moduleSlug="search-console" />;
+	}
+
+	return (
+		<Widget
+			noPadding
+			Header={ Header }
+			Footer={ () => (
+				<SourceLink
+					className="googlesitekit-data-block__source"
+					name={ _x( 'Search Console', 'Service name', 'google-site-kit' ) }
+					href={ serviceURL }
+					external
+				/>
+			) }
```

Comment:
Let's move this into its own `Footer` component the same way that you've done with `Header`.